packages/strapi-admin/admin/src/containers/Admin/reducer.js

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Admin reducer
+ *
+ */
+
+import { fromJS, Map } from 'immutable';
+import {
+  GET_INIT_DATA_SUCCEEDED,
+  HIDE_LEFT_MENU,
+  SET_APP_ERROR,
+  SHOW_LEFT_MENU,
+} from './constants';
+
+const initialState = fromJS({
+  autoReload: false,
+  appError: false,
+  currentEnvironment: 'development',
+  isLoading: true,
+  isSecured: false,
+  layout: Map({}),
+  showMenu: true,
+  strapiVersion: '3',
+  uuid: false,
+});
+
+function adminReducer(state = initialState, action) {
+  switch (action.type) {
+    case GET_INIT_DATA_SUCCEEDED:
+      return state
+        .update('autoReload', () => action.data.autoReload.enabled)
+        .update('currentEnvironment', () => action.data.currentEnvironment)
+        .update('isLoading', () => false)
+        .update('layout', () => Map(action.data.layout))
```

Comment:
I think the layout property used in the leftmenu should be removed since we are going to develop an internal api that will allow to customize it better than with this config

Reply:
We have to be sure that nothing depends on it... Why did we add it?

Reply:
to be able to customize the menu...